### PR TITLE
Added support for records list filtering.

### PR DESCRIPTION
**Pull request by n1k0**
r=? @leplatrem @Natim @almet @magopian (first in served first (please don't fight))

### CODE REVIEW

**Comment by n1k0** [comment]
I'm getting mad at this:

**Comment by n1k0** [comment]
Okay, I realize maybe the filter values needs url encoding or escaping; I'll check that out though if you have hints, that's cool too.

File: test/integration_test.js [hunk 1]
```diff
@@ -18,7 +18,10 @@ const TEST_KINTO_SERVER = "http://0.0.0.0:8888/v1";
 describe("Integration tests", () => {
   let sandbox, server, api;
 
-  before(() => server = new KintoServer(TEST_KINTO_SERVER));
+  before(function() {
+    this.timeout(20000);
```
Comment (almet) [inline test/integration_test.js]: ooh :(

File: test/integration_test.js [hunk 2]
```diff
@@ -698,6 +714,22 @@ describe("Integration tests", () => {
...
+                return coll.batch(batch => {
+                  batch.createRecord({name: "paul", age: 28});
+                  batch.createRecord({name: "jess", age: 54});
+                  batch.createRecord({name: "john", age: 33});
+                });
+              });
+
+              it("should filter records", () => {
+                return coll.listRecords({sort: "age", filters: {min_age: 30}})
+                  .then(records => records.map(x => x.name))
```
Comment (almet) [inline test/integration_test.js]: s/x/r ?

**Comment by almet** [comment]
About the error you're getting: what do you think about having the client do a requests until it actually works (with an API built with promises, that could be a neat wait to avoid timeouts)

File: test/integration_test.js (hunk 1 above)
Comment (leplatrem) [inline test/integration_test.js]: I didn't open any ticket on this. But @n1k0 said that the last versions takes a lot longer to start

**Comment by leplatrem** [comment]
Looks good from here :)

File: test/collection_test.js [hunk 3]
```diff
@@ -424,6 +424,17 @@ describe("Collection", () => {
...
+
+    describe("Filtering", () => {
+      it("should generate the expected filtering query string", () => {
+        coll.listRecords({sort: "x", filters: {min_y: 2, not_z: 3}});
+
+        sinon.assert.calledWithMatch(client.execute, {
+          path: "/buckets/blog/collections/posts/records?min_y=2&not_z=3&_sort=x",
+          headers: {Foo: "Bar", Baz: "Qux"},
+        });
+      });
```
Comment (n1k0) [inline test/collection_test.js]: Add a test without `sort` to ensure the qs is still properly generated.

**Comment by n1k0** [comment]
> About the error you're getting: what do you think about having the client do a requests until it actually works (with an API built with promises, that could be a neat wait to avoid timeouts)

Yeah, that's actually what's done for flushing the server, I should probably port that strategy to server start as well. File #39 about that.

**Pull request merged**